lmnet/lmnet/datasets/dataset_iterator.py

```diff
@@ -220,16 +221,22 @@ class DatasetIterator:
     available_subsets = ["train", "train_validation_saving", "validation"]
 
     """docstring for DatasetIterator."""
-    def __init__(self, dataset, enable_prefetch=False, seed=0):
+    def __init__(self, dataset, enable_prefetch=False, seed=0, batch_pre_fetch=3, num_parallel_process=1):
```

Comment:
About `batch_pre_fetch`, In my feeling, the default `3` odd number looks strange 🤔 Why do you not use `4` ? Is it better setting for your dataset ? (This comment is not request to change)

Reply:
This number will define the queue size resulting the RAM usage. The number 3 is arbitrarily chosen which is suitable for lm-server5 for running 2 training at a time.